### ISSUE TITLE
Fix `selector-no-deprecated` false positives for `::part()` 

### ISSUE THREAD

**Issue by SaekiTominaga** [stylelint/stylelint#9221]
### What minimal example or steps are needed to reproduce the bug?

```css
::part(acronym) {
}
```

Specify a string defined in the variable `deprecatedHtmlTypeSelectors`  in [lib/reference/selectors.mjs](https://github.com/stylelint/stylelint/blob/main/lib/reference/selectors.mjs) as an argument for the `::part()` pseudo-element. e.g. 'acronym', 'content', 'shadow', ...

### What minimal configuration is needed to reproduce the bug?

```js
/** @type {import('stylelint').Config} */
export default {
	rules: {
		'selector-no-deprecated': true,
	},
};
```

### How did you run Stylelint?

```shell
npx stylelint ./*.css
```

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "17.8.0"
}
```

### What did you expect to happen?

The argument for the pseudo-element `::part()` is the value of the HTML `part` attribute, not the name of the HTML element. Therefore, no errors should occur.

### What actually happened?

```
selector-no-deprecated.css
  1:8  ✖  Deprecated selector "acronym"  selector-no-deprecated

✖ 1 problem (1 error, 0 warnings)
  1 error potentially fixable with the "--fix" option.
```

**Comment by jeddy3** [comment]
@SaekiTominaga Thanks for the report and for using the template.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/contributor-guide/rules#fix-a-bug-in-a-rule) in the Contributor guide.

It may just be that we need to use the [`isStandardSyntaxTypeSelector`](https://github.com/stylelint/stylelint/blob/main/lib/utils/isStandardSyntaxTypeSelector.mjs) util:

https://github.com/stylelint/stylelint/blob/84f2c6b75186dc4370b5a7419e26c661fad3acbe/lib/utils/isStandardSyntaxTypeSelector.mjs#L30-L37

i.e. add the following to the rule:

```js
if (!isStandardSyntaxTypeSelector(node)) return;
```
